src/table/src/engine/manager.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
+
+use async_trait::async_trait;
+use common_telemetry::error;
+use snafu::{ensure, OptionExt};
+
+use crate::engine::TableEngineRef;
+use crate::error::{EngineExistSnafu, EngineNotFoundSnafu, Result};
+
+#[async_trait::async_trait]
+pub trait TableEngineManager: Send + Sync {
+    /// returns `error::EngineNotFound` if engine not found
+    fn engine(&self, name: &str) -> Result<TableEngineRef>;
+
+    /// returns `error::EngineExist` if engine exists
+    fn register_engine(&self, name: &str, engine: TableEngineRef) -> Result<()>;
+
+    /// closes all registered engines
+    async fn close(&self) -> Result<()>;
+}
+pub type TableEngineManagerRef = Arc<dyn TableEngineManager>;
+
+/// Simple in-memory table engine manager
+pub struct MemoryTableEngineManager {
+    pub engines: RwLock<HashMap<String, TableEngineRef>>,
+}
+
+impl MemoryTableEngineManager {
+    pub fn new(default_name: &str, engine: TableEngineRef) -> Self {
```

Comment:
The `deault_name` looks it's not necessary, you can retrieve the engine name by `engine.name()`.